haystack/errors.py

```diff
@@ -14,6 +15,7 @@ class HaystackError(Exception):
     """
 
     def __init__(self, message: Optional[str] = None, docs_link: Optional[str] = None):
+        send_custom_event(event=f"{type(self).__name__} raised")
```

Comment:
I checked how this could work. It's a bit difficult because we are here in the `__init__` of the `Exception` so it hasn't been raised yet. What one could do theoretically is access the stack and extract parts of it within the `__init__` of `HaystackError`. That gives you the file and the line of code that raised the error but not the class:
```
import traceback
print(traceback.extract_stack()[-4:-3])
```
However, the line of code wouldn't be of much help. Only the file name could be interesting and there we would need to remove the path (directory names) first. The error name itself, e.g., PipelineSchemaError already is a hint. Let's keep it as is.